@here/harp-olp-utils/lib/OlpDataProvider.ts

```diff
@@ -52,31 +52,27 @@ export class OlpDataProvider implements DataProvider {
      *
      * @returns A promise which is resolved when the connection has been established.
      */
-    connect(): Promise<void> {
+    async connect(): Promise<void> {
         const settings = new OlpClientSettings({
             environment: "here",
             getToken: this.params.getToken
         });
+
         if (this.params.version !== undefined && this.params.version >= 0) {
-            this.m_versionLayerClient = new VersionedLayerClient(
-                HRN.fromString(this.params.hrn),
-                this.params.layerId,
-                settings
-            );
             this.m_catalogVersion = this.params.version;
-            return Promise.resolve();
         } else {
-            return new CatalogClient(HRN.fromString(this.params.hrn), settings)
-                .getLatestVersion(new CatalogVersionRequest())
-                .then(response => {
-                    this.m_versionLayerClient = new VersionedLayerClient(
-                        HRN.fromString(this.params.hrn),
-                        this.params.layerId,
-                        settings
-                    );
-                    this.m_catalogVersion = response;
-                });
+            const latestVersion = await new CatalogClient(
```

Comment:
why await and not just use .then? The function anyway returns a promise.

Reply:
I've simplified original method, as it unnecessarily duplicated chunks of code. and duplication was caused by problems with conditionals when chaining promises with `.then.

As to question, why? Simple i find async/await more clean and robust than `.then` chains.

Reply:
Good point about reducing the duplicated code, and yes it is cleaner in this case.